src/quasi-newton.jl

```diff
@@ -16,6 +16,20 @@ mutable struct LSR1Model{T, S, M <: AbstractNLPModel{T, S}, Meta <: AbstractNLPM
   op::LSR1Operator
 end
 
+mutable struct DiagonalQNModel{T, S, M <: AbstractNLPModel{T, S}, Meta <: AbstractNLPModelMeta{T, S}} <:
+               QuasiNewtonModel{T, S}
+  meta::Meta
+  model::M
+  op::DiagonalQN
```

Comment:
Here I kept the same "structure template" as in `LSR1Model` and `LBFGSModel`, but I was wondering why the field `op` was not a concrete type? Is it because it is more convienient for display or compilation time?

Reply:
Thanks for keeping the same structure, we haven't modified this in a while, so I am not sure. 
Could you test it and add the concrete type if that's better? I guess the `@code_warntype` will be happier with a concrete type.